wandb/sdk/launch/_project_spec.py

```diff
@@ -207,8 +201,53 @@ def compute_command(self, user_parameters):
         return " ".join(command_arr)
 
     @staticmethod
-    def _sanitize_param_dict(param_dict):
+    def _sanitize_param_dict(param_dict: Dict[str, Any]) -> Dict[str, Optional[str]]:
         return {
             (str(key)): (quote(str(value)) if value is not None else None)
             for key, value in param_dict.items()
         }
+
+
+def get_entry_point_command(
+    entry_point: "EntryPoint", parameters: Dict[str, Any]
+) -> List[str]:
+    """
+    Returns the shell command to execute in order to run the specified entry point.
+    :param project: Project containing the target entry point
+    :param entry_point: Entry point to run
+    :param parameters: Parameters (dictionary) for the entry point command
+    """
+    commands = []
+    commands.append(entry_point.compute_command(parameters))
+    return commands
+
+
+def create_project_from_spec(run_spec: Dict[str, Any], api: Api) -> Project:
```

Comment:
nice move this is def better

Reply:
haha this was done out of circular ref necessity